packages/terra-modal/src/Modal.jsx

```diff
@@ -79,8 +83,28 @@ const defaultProps = {
   role: 'document',
 };
 
-/* eslint-disable react/prefer-stateless-function */
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
 class Modal extends React.Component {
+  constructor() {
+    super();
+    this.handleKeydown = this.handleKeydown.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.closeOnEsc) {
```

Comment:
So you recommend always adding the event handler and only trigger onRequestClose when the prop is set?